[WIP] Modified project files default output to be lowercase

dmportella opened the pull request:
The default capitalization of the outputpath was set to `bin\Debug` and `bin\Release`. I changed it so its default to be lowercase.

Closes #375

Let me know if this is all that is needed.

dmportella committed:
Modified project files default output to be lowercase

The default captilization of the outputpath was set to `bin\Debug` and `bin\Release`. I changed it so its default to be lowercase.

Closes #375

dnfclas commented:
Hi __@dmportella__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.dotnetfoundation.org.
    </span>

TTYL, DNFBOT;

dnfclas commented:
__@dmportella__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, DNFBOT;

dsplaisted commented:
Hi @dmportella,

Thanks a bunch for your interest in helping out.

The projects you've modified are used to build templates for Visual Studio but don't actually use the .NET SDK that we're developing in this repo.

To change the defaults for the folder names, you would need to modify the `.targets` and `.props` files in [src/Tasks/Microsoft.NET.Build.Tasks/build](https://github.com/dotnet/sdk/tree/master/src/Tasks/Microsoft.NET.Build.Tasks/build).  Specifically, Microsoft.NET.DefaultOutputPaths.targets [sets the Configuration to "Debug"](https://github.com/dotnet/sdk/blob/23c6ba17b8da999ba9d9d265c2bbb8bb25dedf51/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.DefaultOutputPaths.targets#L29) by default and then [uses the Configuration to construct the OutputPath](https://github.com/dotnet/sdk/blob/23c6ba17b8da999ba9d9d265c2bbb8bb25dedf51/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.DefaultOutputPaths.targets#L35-L36), if it's not already set.

However, I think we were a bit premature in marking this one up for grabs.  I'd like to have [more discussion](https://github.com/dotnet/sdk/issues/375#issuecomment-261164606) on whether switching to lower case is the right thing to do and whether it will affect other tools that expect the capitalized versions.

On top of that, we are probably going to be changing the way configurations are defined (see #350).

So, feel free to update the pull request by updating the MSBuild files that go in the .NET SDK, but understand that we might end up deciding not to take the change, or that we may make some other changes (likely related to #350) which would affect the way this would need to be implemented.

Thanks again!

dmportella commented:
@dsplaisted hey that is why I tagged as *WIP* I knew there was some conversations to be had plus I wasnt sure where I had to make the changes or even test this.

I will make the change to the PR eg. modify the target files and repull.

dmportella commented:
I will hold this pr for now while we discuss the points raised.

dmportella committed:
Default output path now lowercased

Added `ToLowerInvariant()` calls to the intermidiate and default outputpath so Configuration will be default lowercase.

Ref #397

dsplaisted commented on src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.DefaultOutputPaths.targets:
```diff
@@ -32,14 +32,14 @@ Copyright (c) .NET Foundation. All rights reserved.
     <BaseOutputPath Condition="!HasTrailingSlash('$(BaseOutputPath)')">$(BaseOutputPath)\</BaseOutputPath>
-    <OutputPath Condition="'$(OutputPath)' == '' and '$(PlatformName)' == 'AnyCPU'">$(BaseOutputPath)$(Configuration)\</OutputPath>
-    <OutputPath Condition="'$(OutputPath)' == '' and '$(PlatformName)' != 'AnyCPU'">$(BaseOutputPath)$(PlatformName)\$(Configuration)\</OutputPath>
+    <OutputPath Condition="'$(OutputPath)' == '' and '$(PlatformName)' == 'AnyCPU'">$(BaseOutputPath)$(Configuration).ToLowerInvariant()\</OutputPath>
```
This should be `$(Configuration.ToLowerInvariant())` instead of `$(Configuration).ToLowerInvariant()` (and likewise in the other locations).

FYI, you should be able to just run `build` from a command prompt in the repo folder to build and run all the tests.

dmportella replied:
thank you been a while since I did any msbuild @dsplaisted for some reason in linux the build is not working right with a clean checkout it check out the gist for the build out put: https://gist.github.com/dmportella/38d7ded5cd46747afb2f047a944df510

I will push the new changes

dmportella committed:
Added ToLowerInvariant to default output targets.

Updated codebase from build output and review

dmportella commented:
@dsplaisted Yeah i cant build it on my local machine... and on a clean windows virtual machine I have hundreds of errors (no code changes)

dmportella commented:
for some reason i cant see the build logs
![image](https://cloud.githubusercontent.com/assets/676817/20747199/29003d78-b6e1-11e6-8c73-c95c1fd2d29d.png)

And i am logged in

dmportella commented:
@dsplaisted ah the error on my build it seems I needed libicu52 (i had 55)
I will check the build error locally if ic an reproduce it and see if I can fix it might be a test somewhere etc.

dmportella commented:
Seems some of the test libraries are being built in release mode but the sdk is in debug build configuration ... I am doing a release build to see if it works

dmportella commented:
i will try and run the tests in the specific version of ubuntu... i am on elementary os loki (based in xenial ubuntu 16)

dmportella commented:
I created a brand new ubuntu image 14.04 to run the tests.

* I built the master branch and it worked.
* I built my branch with the changes and it is not working.

> build master and then building the branch all works but that because the build doesnt clear the build output from the previous build.

eerhardt commented:
@srivatsn @nguerrera @dsplaisted - Are we still interested in taking this change? I would assume this would be considered a breaking change on case-sensitive file systems.

nguerrera commented:
Yes, it would be a breaking change. I don't think we can just make it the default, but we could provide an option UseLowerCaseDefaultPaths or something. Let's discuss on #375. I'm going to close this stale PR.

dmportella commented:
yea that is fine - i had looked at it and it not quite as simple
 i tried to get the code changes but ended up giving up on the pr
